en/docs/chapter_hashing/hash_collision.md

```diff
@@ -1,108 +1,111 @@
-- **Adding elements**: First access the list head node via the hash function, then add the node (key-value pair) to the list.
-- **Deleting elements**: Access the list head based on the hash function's result, then traverse the list to find and remove the target node.
+- **Querying Elements**: Input `key`, obtain the bucket index through the hash function, then access the head node of the linked list. Traverse the linked list and compare key to find the target key-value pair.
+- **Adding Elements**: Access the head node of the linked list via the hash function, then append the node (key-value pair) to the list.
+- **Deleting Elements**: Access the head of the linked list based on the result of the hash function, then traverse the linked list to find the target node and delete it.
 
 Separate chaining has the following limitations:
 
-- **Increased space usage**: The linked list contains node pointers, which consume more memory space than arrays.
-- **Reduced query efficiency**: Due to the need for linear traversal of the list to find the corresponding element.
+- **Increased Space Usage**: The linked list contains node pointers, which consume more memory space than arrays.
+- **Reduced Query Efficiency**: This is because linear traversal of the linked list is required to find the corresponding element.
+
 
 The code below provides a simple implementation of a separate chaining hash table, with two things to note:
 
 - Lists (dynamic arrays) are used instead of linked lists for simplicity. In this setup, the hash table (array) contains multiple buckets, each of which is a list.
-- This implementation includes a method for resizing the hash table. When the load factor exceeds $\frac{2}{3}$, we resize the hash table to twice its original size.
+- This implementation includes a hash table resizing method. When the load factor exceeds $\frac{2}{3}$, we expand the hash table to twice its original size.
 
 ```src
 [file]{hash_map_chaining}-[class]{hash_map_chaining}-[func]{}
 ```
 
-It's worth noting that when the list is very long, the query efficiency $O(n)$ is poor. **At this point, the list can be converted to an "AVL tree" or "Red-Black tree"** to optimize the time complexity of the query operation to $O(\log n)$.
+It's worth noting that when the linked list is very long, the query efficiency $O(n)$ is poor. **In this case, the list can be converted to an "AVL tree" or "Red-Black tree"** to optimize the time complexity of the query operation to $O(\log n)$.
 
 ## Open addressing
 
-"Open addressing" does not introduce additional data structures but uses "multiple probes" to handle hash collisions. The probing methods mainly include linear probing, quadratic probing, and double hashing.
+"Open addressing" does not introduce additional data structures but instead handles hash collisions through "multiple probing". The probing methods mainly include linear probing, quadratic probing, and double hashing.
```

Comment:
We do not use "but" and "instead" together. I would suggest changing the second part to ", instead it handles ...".